### ISSUE TITLE
Editor crash on MSBuild manual build

### ISSUE THREAD

**Issue by grevius5** [godotengine/godot#51740]
### Godot version

3.4.beta.3 mono

### System information

Windows 10

### Issue description

Editor instant crash when using manual click on "Rebuild Solution". (it occurs 1 time on 10 build clicks on average)

### Steps to reproduce

Select "Rebuild Solution" manually multiple times and all editor will close without showing errors.

**Comment by akien-mga** [comment]
When did you download your 3.4 beta 3 build? The original release had a critical Mono regression which was hotfixed the next day. If you downloaded before Aug 7 at 14:30 UTC, it would be affected.

**Comment by grevius5** [comment]
Yes, I use the 7 Aug version, so i'll wait for the next release to better test that

**Comment by akien-mga** [comment]
I uploaded fixed binaries on Aug 7 at 14:30 UTC so you can try to redownload and see if that solves it: https://downloads.tuxfamily.org/godotengine/3.4/beta3/

**Comment by grevius5** [comment]
I confirm that bug is present in the version that you link to me, 3.4.beta 3 mono (7 Aug 14:30)

**Comment by akien-mga** [comment]
I can't seem to reproduce the issue on Linux with 3.4 beta 3 and the Dodge the Creeps C# demo. Does it happen on any project for you, or only a specific one? Can you see if there are relevant logs in `%APPDATA%/Godot/mono`?

**Comment by grevius5** [comment]
I have the same error in 3.3.2-stable, here the log file to better inspection
[2021-08-19_12.48.46_24796.log](https://github.com/godotengine/godot/files/7013874/2021-08-19_12.48.46_24796.log)

**Comment by 1nZ4N3** [comment]
I also have the same error in 3.3.3 stable. Build log doesn't print any error but in console it print "ERROR: debug_send_unhandled_exception_error: System.NullReferencedException: Object reference not set to an instance of an object
At: modules/mono/mono_gd/gd_mono_utils.cpp:370" for split second before it close.
